sdk/samples/iot/hub/src/pnp_helper.h

```diff
@@ -0,0 +1,137 @@
+ * @param[out] component_name The parsed component name (if it exists).
+ * @param[out] command_name The parsed command name.
+ */
+az_result pnp_helper_parse_command_name(
+    az_span component_command,
+    az_span* component_name,
+    az_span* command_name);
+
+/**
+ * @brief Build a reported property
+ *
+ * @param[in] json_buffer The span into which the json payload will be placed.
+ * @param[in] component_name The name of the component for the reported_property.
+ * @param[in] property_name The name of the property to which to send an update.
+ * @param[in] append_callback The user callback to invoke to add the property value.
+ * @param[in] context The user context which is passed to the callback.
+ * @param[out] out_span The #az_span pointer to the output json payload.
+ */
+az_result pnp_helper_create_reported_property(
+    az_span json_buffer,
+    az_span component_name,
+    az_span property_name,
+    pnp_append_property_callback append_callback,
+    void* context,
+    az_span* out_span);
+
+/**
+ * @brief Build a reported property
+ *
+ * @param[in] json_buffer The span into which the json payload will be placed.
+ * @param[in] component_name The name of the component for the reported_property.
+ * @param[in] property_name The name of the property to which to send an update.
+ * @param[in] append_callback The user callback to invoke to add the property value.
+ * @param[in] context The user context which is passed to the callback.
+ * @param[in] ack_code The return value for the reported property.
+ * @param[in] ack_version The ack version for the reported property.
+ * @param[in] ack_description The optional description for the reported property.
+ * @param[out] out_span The #az_span pointer to the output json payload.
+ */
+az_result pnp_helper_create_reported_property_with_status(
+    az_span json_buffer,
+    az_span component_name,
+    az_span property_name,
+    pnp_append_property_callback append_callback,
+    void* context,
+    int32_t ack_code,
+    int32_t ack_version,
+    az_span ack_description,
+    az_span* out_span);
+
+/**
+ * @brief Iteratively get the next desired property.
+ *
+ * @param[in] json_reader A pointer to the json reader from which the properties will be retrieved.
+ * @param[in] is_partial Boolean stating whether the JSON document is partial or not.
+ * @param[in] sample_components_ptr A pointer to a set of `az_span` pointers containing all the
+ * names for components.
+ * @param[in] sample_components_num Number of components in the set pointed to by
+ * `sample_components_ptr`.
+ * @param[in] scratch_buf Scratch buffer to place parsed JSON values.
+ * @param[in] scratch_buf_len Length of `scratch_buf`.
+ * @param[in] property_callback The callback which is called on each twin property.
+ * @param[in] context_ptr Pointer to user context.
+ */
+az_result pnp_helper_process_twin_data(
+    az_json_reader* json_reader,
+    bool is_partial,
+    const az_span** sample_components_ptr,
```

Comment:
or ?
```suggestion
    const az_span* sample_components_ptr[],
```

Reply:
by convention elsewhere (v1 etc) it's usually been a double pointer. i'll probably keep as is for now unless someone has a good reason to switch